crates/runtime/src/dataconnector/s3.rs

```diff
@@ -131,6 +177,15 @@ impl DataConnectorFactory for S3Factory {
                 }
             }
 
+            if let Some(region) = params.parameters.get("region").expose().ok() {
+                if !AWS_REGIONS.contains(&region) {
```

Comment:
Can region be defined upper case / need to lowercase the region while checking?

Reply:
good call, region cannot be uppercase. I've added a new error type to warn when we automatically convert it to lower case:

```bash
2024-11-20T05:52:48.718016Z  WARN runtime::dataconnector::s3: The `s3_region` parameter requires a lowercase AWS region code, but 'US-EAST-1' was provided.
Spice will automatically convert the region code to lowercase.
For further information, visit: https://docs.spiceai.org/components/data-connectors/s3#params
```